gnocchi/amqp1d.py

```diff
@@ -0,0 +1,218 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+import itertools
+import uuid
+
+import daiquiri
+import proton.handlers
+import proton.reactor
+import six
+import ujson
+
+from gnocchi import incoming
+from gnocchi import indexer
+from gnocchi import service
+from gnocchi import utils
+
+LOG = daiquiri.getLogger(__name__)
+
+
+class BatchProcessor(object):
+    def __init__(self, conf):
+        self.conf = conf
+        self.incoming = incoming.get_driver(self.conf)
+        self.indexer = indexer.get_driver(self.conf)
+        self._ensure_resource_type_exists()
+
+        self._hosts = {}
+        self._measures = collections.defaultdict(
+            lambda: collections.defaultdict(list))
+
+    def reset(self):
+        self._hosts.clear()
+        self._measures.clear()
+
+    def add_measures(self, host, name, measures):
+        host_id = "%s:%s" % (self.conf.amqp1d.resource_type,
+                             host.replace("/", "_"))
+        self._hosts[host_id] = host
+        self._measures[host_id][name].extend(measures)
+
+    def flush(self):
+        try:
+            self._flush()
+        except Exception:
+            LOG.error("Unepected error during flush()", exc_info=True)
+        self.reset()
+
+    def _flush(self):
+        archive_policies = {}
+        for host_id, measures_by_names in six.iteritems(self._measures):
+            resource = self._get_resource(host_id)
```

Comment:
I don't really like the fact it gets resource one by one but I guess that's ok for now.

Reply:
fixed.